File: cbam/cbam/web_form/add_operating_company_data/add_operating_company_data.js




frappe.ready(function() {
	frappe.call({
		method: "cbam.cbam.web_form.add_operating_company_data.add_operating_company_data.get_country_codes",
		callback: (r) => {
			if (r.message.countryCodesOptions && r.message.countryCodesOptions.length > 0) {
				frappe.web_form.fields_dict["country_code"]._data = r.message.countryCodesOptions;
			}
		}
	});
	
	frappe.web_form.after_save = () => {
		console.log("OKk")
		console.log(frappe.web_form.doc.name)
		frappe.call({
			method: "cbam.cbam.web_form.add_operating_company_data.add_operating_company_data.setup_new_employee",
			args: {
				"parent_supplier":frappe.web_form.get_value("parent_supplier"),
				"email": frappe.web_form.get_value("email")
		}
	});
}
});

File: cbam/cbam/web_form/add_operating_company_data/add_operating_company_data.py
import frappe

def get_context(context):
	pass

@frappe.whitelist()
def get_country_codes():
	try:
		country_code_list = frappe.get_all("Country Code", fields=["name", "country_full_name"])
		country_code_options = [{"label": c.country_full_name, "value": c.name} for c in country_code_list]
		return {
			"countryCodesOptions": country_code_options,
		}
	except:
		frappe.throw("Couldn't create a valid selection for the Field 'Country'.")


@frappe.whitelist()
def setup_new_employee(email):
	supplier = frappe.db.get_value("Supplier Employee Item", {"employee_email": frappe.session.user}, "parent")
	doc = frappe.get_doc("CBAM Operating Company", {"email":email, "parent_supplier": supplier})
	doc._create_new_employee()
	doc.save()